File: JobBoardService/src/main/java/com/project4/JobBoardService/DTO/CompanyDTO.java
package com.project4.JobBoardService.DTO;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.util.List;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class CompanyDTO {

    private Long companyId;


    private String companyName;


    private String logo;


    private String websiteLink;


    private String description;




    private String location;
    private String type;

    private String keySkills;
    private String companySize;
    private String country;
    private String countryCode;
    private String workingDays;
    private List<ReviewDTO> reviews;  // List of ReviewDTO

    private List<JobApplicationDTO> jobApplications; // Chỉ bao gồm thông tin cần thiết
    private Boolean membershipRequired = false;
    public void setReviews(List<ReviewDTO> reviews) {
        this.reviews = reviews;
    }

    public void setJobs(List<JobDTO> jobDTOs) {

    }
}

File: jobboardmobile/lib/dto/company_dto.dart
class CompanyDTO {
  final int companyId;
  final String companyName;
  final String logo;
  final String websiteLink;
  final String description;
  final String location;
  final String keySkills;
  final String type;
  final String companySize;
  final String country;
  final String countryCode;
  final String workingDays;
  final bool membershipRequired;

  CompanyDTO({
    required this.companyId,
    required this.companyName,
    required this.logo,
    required this.websiteLink,
    required this.description,
    required this.location,
    required this.keySkills,
    required this.type,
    required this.companySize,
    required this.country,
    required this.countryCode,
    required this.workingDays,
    required this.membershipRequired,
  });

  factory CompanyDTO.fromJson(Map<String, dynamic> json) {
    return CompanyDTO(
      companyId: json['companyId'],
      companyName: json['companyName'],
      logo: json['logo'],
      websiteLink: json['websiteLink'],
      description: json['description'],
      location: json['location'],
      keySkills: json['keySkills'],
      type: json['type'],
      companySize: json['companySize'],
      country: json['country'],
      countryCode: json['countryCode'],
      workingDays: json['workingDays'],
      membershipRequired: json['membershipRequired'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'companyId': companyId,
      'companyName': companyName,
      'logo': logo,
      'websiteLink': websiteLink,
      'description': description,
      'location': location,
      'keySkills': keySkills,
      'type': type,
      'companySize': companySize,
      'country': country,
      'countryCode': countryCode,
      'workingDays': workingDays,
      'membershipRequired': membershipRequired,
    };
  }
}
